SQL Editor: CLI commands to import snippet as migration/seed

gregnr opened the pull request:
### 🔍 Preview
https://studio-staging-git-feat-sql-editor-cli-import-supabase.vercel.app/dashboard/project/_/sql/new

 
Adds new button "Download as migration file" to SQL editor in the snippet context menu:

 ![screenshot-2023-11-16-at-14 49 37](https://github.com/supabase/supabase/assets/105593/dbb32c6c-ca17-4715-97f2-9c55715dafc7)


Clicking the button brings up a modal with pre-generated commands to import the current snippet as a migration, seed, or SQL file:
<img width="796" alt="screenshot-2023-11-16-at-14 49 57" src="https://github.com/supabase/supabase/assets/105593/75d0751b-824b-47ea-bdea-05bbe334439c">
<img width="813" alt="screenshot-2023-11-16-at-14 49 49" src="https://github.com/supabase/supabase/assets/105593/34f531eb-fde7-47b0-a040-b4a58e32dbb7">

gregnr committed:
feat(sql-editor): cli commands to import snippet as migration/seed

gregnr committed:
fix(sql-editor): reword to download snippet

gregnr committed:
chore(sql-editor): update wording

gregnr committed:
Merge branch 'master' into feat/sql-editor-cli-import

joshenlim commented on studio/components/interfaces/SQLEditor/UtilityPanel/UtilityActions.tsx:
```diff
@@ -23,10 +26,31 @@ const UtilityActions = ({
+          icon={<IconDownload size="tiny" />}
+          onClick={() => setIsDownloadSnippetModalOpen(true)}
+        >
+          Download as migration file
```
i'd opt to put "Download as migration file" behind a tooltip to prevent clutter here

same considerations for table editor header - lets keep things neat and clean while we still can

joshenlim replied:
Can i also check if this is a heavily requested feature? Alternatively we put this in the snippet's dropdown in the left side nav

gregnr replied:
Good points @joshenlim - how about we put it under the snippet's left side dropdown for now and we can make it more visible in the future if there's lots of usage

joshenlim commented:
This is just future consideration - but we could consider leveraging the new API side panel for this too
have a section for SQL editor
clicking "Download as migration" opens the API side panel with that section open

joshenlim commented:
was also looking at the thread - i was thinking the same thing with Terry RE tabs, could look something like this
![image](https://github.com/supabase/supabase/assets/19742402/039c6cc5-7038-4ca0-9042-faa96ecb6a8c)

language selection as a dropdown on the side
main consideration is that users are probably looking for one thing at a time, its a bit of a spread showing all three types at once

saltcod committed:
Merge branch 'master' into feat/sql-editor-cli-import

saltcod committed:
Move to tabed ui, move download button to context menu

saltcod commented:
Hey guys – made some updates here. Sorry for the delay.

1. moved the button to the context menu for ui simplicity 
2. changed the layout to a tab, avoiding tab-ception via the toggle to the right 

no major changes otherwise
<img width="854" alt="screenshot-2023-11-13-at-11 25 19" src="https://github.com/supabase/supabase/assets/105593/753b8b1a-67f6-44dc-98a0-1ce0924671f6">

<img width="363" alt="screenshot-2023-11-13-at-11 25 13" src="https://github.com/supabase/supabase/assets/105593/5188ce6d-9f88-4458-8cb0-66f6ad6e19e3">

saltcod committed:
Inline

saltcod committed:
Merge branch 'master' into feat/sql-editor-cli-import

thorwebdev committed:
feat: add react native auth blog and quickstart. (#19006)

* feat: add react native auth blog and quickstart.

* Grammarly check

* Minor edits.

* Added og image.

* chore: add password sign in and gif.

---------

Co-authored-by: Ramiro Nuñez Dosio <ramiro@supabase.io>

kevcodez committed:
fix: duplicate eslint dependency (#19028)

uncapped1599 committed:
Update sveltekit.mdx (#18998)

Extended authorization section of docs to mention use of Sveltekit sequence helper function for protecting routes alongside defining Supabase session. Also provided new code.

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

laktek committed:
fix: add hello world example to edge functions quickstart (#19030)

joshenlim committed:
Fix project card status not showing (#19034)

* Fix project card status not showing

* Fix tooltip project card styling

kiwicopple committed:
chore: update cli reference doc (#18934)

kiwicopple committed:
Add docs for updating the image versions (#18160)

see #17930

darora committed:
chore: fix casing of "encryption at rest" (#19032)

kiwicopple committed:
chore: update self-hosted image versions (#18935)

chore: update image versions for docker/docker-compose.yml

olirice committed:
(docs) Update arg name that resulted from adapters release (#19024)

update arg name that resulted from adapters release

taishikato committed:
docs: Fix an example (@supabase/ssr package) for Next.js API Route (#19009)

Update creating-a-client.mdx

replace `appendHeader` with `setHeader` for the Next.js API route example.

kevcodez committed:
chore: cleanup studio dependencies (#19040)

fsansalvadore committed:
Fix code text color (#19042)

* make dark: selector work again

* fix reference docs code color

inian committed:
feat: remove note about tus rollout (#19007)

it has been rolled out to all projects

joseph082 committed:
docs: update language in database-size.mdx code block (#19033)

Co-authored-by: Kevin Grüneberg <k.grueneberg1994@gmail.com>
Co-authored-by: Copple <10214025+kiwicopple@users.noreply.github.com>

kiwicopple committed:
Docs: Adds a page for enums (#18590)

* Adds a page for enums

* Adds some details on deleting values

* Update apps/docs/pages/guides/database/postgres/enums.mdx

Co-authored-by: Kevin Grüneberg <k.grueneberg1994@gmail.com>

* Update apps/docs/pages/guides/database/postgres/enums.mdx

Co-authored-by: Kevin Grüneberg <k.grueneberg1994@gmail.com>

* Update apps/docs/pages/guides/database/postgres/enums.mdx

Co-authored-by: Kevin Grüneberg <k.grueneberg1994@gmail.com>

* Update apps/docs/pages/guides/database/postgres/enums.mdx

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

* removes dangerous operation

---------

Co-authored-by: Kevin Grüneberg <k.grueneberg1994@gmail.com>
Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

maidi29 committed:
Update hugging-face.mdx: fix example curl request (#17667)

The example request did not match the example edge function because there is used prompt and in the curl script was used query

morinokami committed:
docs: fix syntax error in create role command (#17755)

abs committed:
Update auth-sso-saml.mdx (#17856)

abotsi committed:
docs: Update auth-google.mdx (#18119)

filipecabaco committed:
fix: Bump up Realtime to 2.25.35 (#19048)

thinkjrs committed:
Update llamaindex.ipynb doc_hash to hash (#18838)

* Update llamaindex.ipynb doc_hash to hash

Update `doc_hash` to `hash` since running this as-is provided an `AttributeError`. 

Here's what the Python build-in `vars` gives me (truncated):

```python
{'id_': 'd94b1287-b3ef-4aba-ac6b-7a4e353b0327',
 'embedding': None,
 'metadata': {},
 'excluded_embed_metadata_keys': [],
 'excluded_llm_metadata_keys': [],
 'relationships': {},
 'hash': '4c702b4df575421e1d1af4b1fd50511b226e0c9863dbfffeccb8b689b8448f35',...}
```

* fix: SimpleWebPageReader & StorageContext imports

* chore: revert python version change

---------

Co-authored-by: Greg Richardson <greg.nmr@gmail.com>

charislam committed:
fix: add reference docs back to search index (#19054)

Reference docs weren't being indexed properly because the specs weren't
getting checked out at build time.

thorwebdev committed:
docs: add RN videos and discoverability improvements. (#18633)

* docs: add RN videos and discoverability improvements.

* Apply suggestions from code review

Co-authored-by: Copple <10214025+kiwicopple@users.noreply.github.com>

* chore: run prettier.

---------

Co-authored-by: Copple <10214025+kiwicopple@users.noreply.github.com>

Commit:
docs: add getting started guide for Swift (#19044)

* docs: getting started guide for Swift

* Apply suggestions from code review

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

* Add new tutorial to menu

* Add correct image for tutorial

* style: format with-swift tutorial

* docs: add example to the repository

---------

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>
Co-authored-by: Thor 雷神 Schaeff <5748289+thorwebdev@users.noreply.github.com>

dshukertjr committed:
docs: Update mentions of `foreign table` to `referenced table` (#18656)

* replace foreignTable parameter with referencedTable parameter

* remove redirects

* rename any mentions of foreign table to referenced table

* fix one mention of foreign table

* update some links

dshukertjr committed:
guide: add pgroonga videos to pgroonga guide (#19073)

add pgroonga videos to pgroonga guide

Commit:
docs(swift): rename `authStateChanges` and remove `File` from upload method (#19067)

docs: fix authStateChanges and upload documentation

joshenlim commented on apps/studio/components/layouts/SQLEditorLayout/QueryItem.tsx:
```diff
@@ -155,6 +158,8 @@ const QueryItemActions = observer(({ tabInfo, activeId }: QueryItemActionsProps)
     setDeleteModalOpen(true)
   }
 
+  const onClickDownloadMigration = () => {}
```
@gregnr missing implementation here?

joshenlim commented on apps/studio/components/interfaces/SQLEditor/DownloadSnippetModal.tsx:
```diff
@@ -0,0 +1,150 @@
+              <div className="flex flex-col gap-2 my-2 w-full">
+                <h2 className="text-lg">Download as migration</h2>
+                <p className="text-sm text-scale-1000">
+                  Use the snippet in a new migration named{' '}
```
just wondering - should we also just state where to run this command in? help the user with less thinking

also, am not too sure what the outcome of this command is but is `use` the right verb here? should it be `download`? "Download the snippet into a new migration"?

saltcod replied:
1. changed 'use' to 'download'. I think that's right. 
2. added links at the bottom here. Played with adding helper text above the tabs there as well, but felt like it was getting in the way. 
Do you think this is enough? The "learn more about" should convey:  "this is for local development", which may or may not be pertinent to you. Does it convey that well enough to you? 

<img width="870" alt="screenshot-2023-11-20-at-10 24 34" src="https://github.com/supabase/supabase/assets/105593/a0302d7a-c909-4c05-bc75-333c181e6f3e">

saltcod replied:
![screenshot-2023-11-20-at-10 34 10](https://github.com/supabase/supabase/assets/105593/be28b873-9279-4153-89bf-ceb0c09f4b97)

joshenlim commented on apps/studio/components/interfaces/SQLEditor/DownloadSnippetModal.tsx:
```diff
@@ -0,0 +1,150 @@
+              <div className="flex flex-col gap-2 my-2 w-full">
+                <h2 className="text-lg">Download as seed file</h2>
+                <p className="text-sm text-scale-1000">
+                  Alternatively if your query consists of sample data, append the snippet to the end
```
am just wondering if this should make the outcome clearer

```suggestion
                  Alternatively if your query consists of sample data, run the following command to append the snippet to the end
```

joshenlim replied:
I don't think we should add "Alternatively" here either - we should make each tab mutually exclusive i think language wise, although I get what we're trying to do here where the user starts from the first tab, then goes to the second

joshenlim committed:
Small style fixes

joshenlim committed:
Merge branch 'master' into feat/sql-editor-cli-import

joshenlim committed:
Style

saltcod committed:
Merge branch 'master' into feat/sql-editor-cli-import

saltcod committed:
Merge branch 'master' into feat/sql-editor-cli-import

saltcod committed:
Update

gregnr commented on apps/studio/components/interfaces/SQLEditor/DownloadSnippetModal.tsx:
```diff
@@ -141,7 +120,43 @@ const DownloadSnippetModal = ({ id, ...props }: DownloadSnippetModalProps) => {
               )}
             </Tabs.Panel>
+            <p className="text-xs text-lighter mt-4 text-right mr-6 mb-4">
+              * Run from your local <code>/supabase</code> directory
```
This should be project root right? Otherwise we should update each command to remove the `./supabase` prefix.

saltcod committed:
Update

joshenlim committed:
Update UI components positioning for DownloadSnippetModal
